models/src/main/scala/coop/rchain/models/rholangn/parmanager/RhoHash.scala

```diff
@@ -0,0 +1,384 @@
+package coop.rchain.models.rholangn.parmanager
+
+import coop.rchain.models.rholangn._
+import coop.rchain.models.rholangn.parmanager.Constants._
+import org.bouncycastle.crypto.digests.Blake2bDigest
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.annotation.unused
+
+private[parmanager] object RhoHash {
```

Comment:
I have a feeling this can be simplified private class and private object to have one public function looks excessive, but let's return to this when we get to refactoring.